test/sinatra/ssl_client_verification_integration_test.rb

```diff
@@ -0,0 +1,78 @@
+require 'test_helper'
+require 'net/http'
+
+class SSLClientVerificationIntegrationTest < Test::Unit::TestCase
+  include Proxy::IntegrationTestCase
+
+  class TestAPIWithSSLClientAuth < ::Sinatra::Base
+    helpers ::Proxy::Helpers
+    authorize_with_ssl_client
+    get('/') { 'Success' }
+  end
+
+  class TestPluginWithSSLClientAuth < ::Proxy::Plugin
+    class << self
+      def http_rackup
+        'run SSLClientVerificationIntegrationTest::TestAPIWithSSLClientAuth'
+      end
+      alias https_rackup http_rackup
+    end
+  end
+
+  def test_http
+    launch protocol: 'http', plugins: [TestPluginWithSSLClientAuth]
+    res = Net::HTTP.get_response('localhost', '/', @settings.http_port)
+    assert_kind_of Net::HTTPSuccess, res
+    assert_equal 'Success', res.body
+  end
+
+  def test_https_no_cert
+    launch_https
+    http = Net::HTTP.new('localhost', @settings.https_port)
+    http.use_ssl = true
+    http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+    res = http.get('/')
+    assert_kind_of Net::HTTPForbidden, res
+  end
+
+  def test_https_wrong_cert
```

Comment:
Thanks, I've updated the name of the test to state that the cert's from a different authority.